packages/core/lib/commands/compile.js

```diff
@@ -97,10 +97,10 @@ const command = {
   },
 
   listVersions: async function (options) {
-    const { CompilerSupplier } = require("@truffle/compile-solidity");
+    const { createCompilerSupplier } = require("@truffle/compile-solidity");
     const { asyncTake } = require("iter-tools");
 
-    const supplier = new CompilerSupplier({
+    const supplier = createCompilerSupplier({
       solcConfig: {
         ...options.compilers.solc,
```

Comment:
As @haltman-at [mentions](https://github.com/trufflesuite/truffle/pull/4222#discussion_r687283884), this will cause problems if `version` is set to `"native"` or to a local soljson path.

This should probably explicitly set `version: undefined` to ensure VersionRange or Docker get used.